pkg/lib/template/template_loader.go

```diff
@@ -173,6 +177,59 @@ func (loader *Loader) Render(
 			}
 			return data, err
 		},
+		"file_sqlite": func(path, statement string) ([]map[string]interface{}, error) {
+			sqliteFile := filepath.Join(rootDir, path)
+			database, err := sql.Open("sqlite3", sqliteFile)
+			if err != nil {
+				return nil, err
+			}
+			defer database.Close()
+
+			rows, err := database.Query(statement)
+			if err != nil {
+				return nil, err
+			}
+			defer rows.Close()
+
+			columns, err := rows.ColumnTypes()
+			if err != nil {
+				return nil, err
+			}
+
+			data := []map[string]interface{}{}
+
+			for rows.Next() {
+				row := []interface{}{}
+				dataEntry := map[string]interface{}{}
```

Comment:
You can use `make` here to allocate the memory for the data. So you don't need to use append below.